### ISSUE TITLE
w3af boot problem

### ISSUE THREAD

**Issue by betacrak** [andresriancho/w3af#17917]
w3af boot problem
I have a problem starting w3af. the following error message appears when launching the following script: ./ w3af_gui.
What should be done?.
Regards,
Diop

[configw3af.docx](https://github.com/andresriancho/w3af/files/3269985/configw3af.docx)

**Comment by andresriancho** [comment]
Please include the error in the github issue. Not going to download and open a `docx` file.

**Comment by betacrak** [comment]
Hi Mr Riancho,
Here you go
Contents of /etc/lsb-release
---------------------------------
DISTRIB_ID=Kali
DISTRIB_RELEASE=kali-rolling
DISTRIB_CODENAME=kali-rolling
DISTRIB_DESCRIPTION="Kali GNU/Linux Rolling"
---------------------------------------------------
Contents of /tmp/w3af_dependency_install.sh

npm install -g retire
npm update -g retire

---------------------------------------------------------
Output pip freeze

acora==2.1
adns-python==1.2.1
anyjson==0.3.3
argcomplete==1.8.1
asn1crypto==0.24.0
attrs==19.1.0
Automat==0.6.0
backdoor-factory==0.0.0
backports-abc==0.5
backports.shutil-get-terminal-size==1.0.0
backports.ssl-match-hostname==3.5.0.1
BBQSQL==1.0
bdfproxy==0.0.0
BeautifulSoup==3.2.1
beautifulsoup4==4.6.0
biplist==1.0.3
BlindElephant==1.0
blinker==1.3
bravado-core==5.12.1
capstone==3.0.4
certifi==2017.11.5
chardet==3.0.4
chirp===daily-20170714
click==6.7
cluster==1.1.1b3
colorama==0.3.7
ConfigArgParse==0.11.0
configobj==5.0.6
configparser==3.5.0
constantly==15.1.0
construct==2.8.16
couchdbkit==0.6.5
cryptography==2.1.4
cycler==0.10.0
darts.util.lru==0.5
decorator==4.1.2
dicttoxml==1.7.4
diff-match-patch==20121119
distorm3==3.3.4
dnslib==0.9.7
dnspython==1.15.0
docutils==0.14
ds-store==1.1.2
easygui==0.96
Elixir==0.7.1
enum34==1.1.6
esmre==0.3.1
feedparser==5.2.1
Flask==0.10.1
functools32==3.2.3.post2
fuse-python==0.2.1
future==0.15.2
futures==3.2.0
GDAL==2.2.2
GeoIP==1.3.2
gevent==1.2.2
gitdb2==2.0.5
GitPython==2.1.3
greenlet==0.4.12
guess-language==0.2
gyp==0.1
h2==2.5.2
halberd==0.2.4
hpack==3.0.0
html2text==2016.9.19
html5lib==0.999999999
http-parser==0.8.3
httplib2==0.11.3
httpretty==0.8.14
hyperframe==4.0.1
hyperlink==17.3.1
idna==2.6
impacket==0.9.15
incremental==16.10.1
ipaddress==1.0.17
ipaddresses==0.0.2
IPy==0.83
ipython==5.5.0
ipython-genutils==0.2.0
itsdangerous==0.24
jdcal==1.0
Jinja2==2.10
jsbeautifier==1.6.4
jsonpickle==0.9.5
jsonpointer==2.0
jsonref==0.2
jsonrpclib==0.1.7
jsonschema==3.0.1
keepnote==0.7.8
keyring==10.5.1
keyrings.alt==2.2
killerbee==1.0
lxml==3.7.1
lz4==1.1.0
M2Crypto==0.27.0
mac-alias==2.0.7
Mako==1.0.7
Markdown==2.6.1
MarkupSafe==1.0
matplotlib==2.0.0
mechanize==0.2.5
mercurial==4.4.1
metaconfig==0.1.4a1
mitmproxy==0.13
mockito==0.5.2
msgpack==0.5.6
msgpack-python==0.5.6
mysqlclient==1.3.10
nassl==0.12
ndg-httpsclient==0.3.3
netaddr==0.7.19
netlib==0.13.1
NfSpy==1.0
nltk==3.0.1
numpy==1.13.3
olefile==0.44
openpyxl==2.3.0
PAM==0.4.2
paramiko==2.0.0
passlib==1.7.1
pathlib2==2.3.0
pcapy==0.10.8
pdfminer==20140328
Pebble==4.3.8
peepdf==0.3.6
pefile==2017.11.5
pexpect==4.2.1
phply==0.9.1
pickleshare==0.7.4
Pillow==4.3.0
ply==3.11
prettytable==0.7.2
prompt-toolkit==1.0.14
psutil==5.4.8
psycopg2==2.7.3
py==1.4.34
pyasn1==0.4.2
pyasn1-modules==0.0.7
pybloomfiltermmap==0.3.15
pycairo==1.15.4
pyClamd==0.3.15
pycrypto==2.6.1
pycurl==7.43.0
pydns==2.3.6
pyenchant==2.0.0
PyGithub==1.21.0
Pygments==2.2.0
pygobject==3.26.1
pygtkspellcheck==4.0.5
pyinotify==0.9.6
PyJWT==1.5.3
pymongo==3.5.1
pymssql==2.1.3
pyOpenSSL==16.2.0
pyparsing==2.1.10
PyPDF2==1.26.0
pyperclip==1.6.0
pyrit==0.5.1
pyrsistent==0.15.2
pyscard==1.9.6
pyserial==3.4
pysnmp==4.3.2
pysnmp-apps==0.3.2
pysnmp-mibs==0.1.3
PySocks==1.6.5
pyspatialite==3.0.1
pysqlite==2.7.0
pytest==3.2.1
python-dateutil==2.6.1
python-Levenshtein==0.12.0
python-ntlm==1.0.1
pythonaes==1.0
pytz==2017.2
pyusb==1.0.0b2
PyX==0.12.1
pyxdg==0.25
PyYAML==3.12
qrcode==5.3
requests==2.18.4
restkit==4.2.2
rfc3987==1.3.8
rfidiot==1.0
roman==2.0.0
ruamel.ordereddict==0.4.8
scandir==1.6
scapy==2.4.0
SecretStorage==2.3.1
service-identity==16.0.0
Shapely==1.6.3
simplegeneric==0.8.1
simplejson==3.13.2
singledispatch==3.4.0.3
six==1.11.0
slowaes==0.1a1
smmap2==2.0.5
socketpool==0.5.3
SQLAlchemy==1.1.11
strict-rfc3339==0.7
subprocess32==3.5.4
swagger-spec-validator==2.4.3
tblib==0.2.0
tcpwatch==1.3.1
termcolor==1.1.0
tldextract==1.7.2
tornado==4.5.2
traitlets==4.3.2
Twisted==17.9.0
typing==3.6.2
unicodecsv==0.14.1
urllib3==1.22
urwid==1.3.1
uTidylib==0.3
vinetto==0.7b0
volatility==2.6
vulndb==0.1.1
vulners==1.3.0
wafw00f==0.9.3
wapiti==2.3.0
wcwidth==0.1.7
webcolors==1.9.1
webencodings==0.5
webunit==1.3.10
Werkzeug==0.13
wfuzz==2.2.3
Whoosh==2.7.4
wxPython==3.0.2.0
wxPython-common==3.0.2.0
xdot==0.6
XlsxWriter==0.9.6
xmlbuilder==1.0
yara-python==3.7.0
zenmap==7.60
zim==0.67
zope.interface==4.3.2
-------------version ----------
python --version: python 2.7.14+

------------------------------
**Message error w3af when it starts**

root@Pentest:~/w3af# ./w3af_gui
External programs used by w3af are not installed or were not found.Run these commands to install them on your system:

    npm install -g retire
    npm update -g retire

A script with these commands has been created for you at /tmp/w3af_dependency_install.sh
----------------------------------------------------------
I executed the script "/tmp/w3af_dependency_install.sh" but the same error message  appears.
How can I do for starting w3af?????

**Comment by andresriancho** [comment]
Try with `npm install -g retire@2.0.3`

**Comment by betacrak** [comment]
Hi Riancho,
This is  result of command : npm install -g retire@2.0.3
root@Pentest:~# cd w3af/
root@Pentest:~/w3af# ls
circle.yml  extras    README.md  scripts  w3af      w3af_console
doc         profiles  result     tools    w3af_api  w3af_gui


root@Pentest:~/w3af# npm install -g retire@2.0.3
npm WARN npm npm does not support Node.js v10.15.2
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6,
7, 8, 9.
npm WARN npm You can find the latest version at https://nodejs.org/
/usr/local/bin/retire -> /usr/local/lib/node_modules/retire/bin/retire
+ retire@2.0.3
added 1 package from 2 contributors and updated 2 packages in 11.452s


root@Pentest:~/w3af# ./w3af_gui
Traceback (most recent call last):
  File "./w3af_gui", line 110, in <module>
    _main()
  File "./w3af_gui", line 106, in _main
    sys.exit(main())
  File "./w3af_gui", line 100, in main
    from w3af.core.ui.gui.main import main as gui_main
  File "/root/w3af/w3af/core/ui/gui/main.py", line 54, in <module>
    from w3af.core.ui.gui import scanrun, helpers, profiles, compare
  File "/root/w3af/w3af/core/ui/gui/scanrun.py", line 27, in <module>
    import webkit
ImportError: No module named webkit
root@Pentest:~/w3af#

Cordially


Le ven. 14 juin 2019 à 14:38, Andres Riancho <notifications@github.com> a
écrit :

**Comment by betacrak** [comment]
Hi Mr Riancho
I tried the command npm install -g ***@***.*** but don't work 
here you go the result:
root@Pentest:~/w3af# npm install -g ***@******
npm WARN npm npm does not support Node.js v10.15.2
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8, 9.
npm WARN npm You can find the latest version at https://nodejs.org/
npm ERR! code E404
npm ERR! 404 Not Found: ***@******

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-07-03T15_31_29_918Z-debug.log
What can I Do
Cordially

**Comment by arkangel777** [comment]
This is what I am up to in the w3af install on Parrot OS. When I run the ./w3af_console, this is what I get. Does anyone have any solutions please. Thanks.

#./w3af_console 
/usr/local/lib/python2.7/dist-packages/OpenSSL/_util.py:6: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography.hazmat.bindings.openssl.binding import Binding
Error while reading plugin options: "invalid syntax (common.py, line 31)"

**Comment by daqing526533** [comment]
> This is what I am up to in the w3af install on Parrot OS. When I run the ./w3af_console, this is what I get. Does anyone have any solutions please. Thanks.
> 
> #./w3af_console /usr/local/lib/python2.7/dist-packages/OpenSSL/_util.py:6: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release. from cryptography.hazmat.bindings.openssl.binding import Binding Error while reading plugin options: "invalid syntax (common.py, line 31)"


For my case, it is because the latest version of python module swagger_spec_validator is written for python3. Therefore, I solved the proble by downgrade it to 2.7.2 with below two commands:
```
pip uninstall swagger_spec_validator
pip install swagger_spec_validator==2.7.2
```
